cmake/configure.cmake

```diff
@@ -68,6 +68,25 @@ else()
     if(NOT CUDNN_FOUND)
         message(FATAL_ERROR "Paddle need cudnn to compile")
     endif()
+    if (NOT NCCL_INCLUDE_DIR)
```

Comment:
Thanks for teaching me the logic here in our offline discussion @reyoung . I think we can shorten L71-L82 into 

```cmake
if (NOT NCCL_INCLUDE_DIR)
  message(FATAL_ERROR "Paddle needs the NCCL header file")
endif()
```

Reply:
Done.